source/mbed_trace.c

```diff
@@ -330,10 +378,11 @@ void mbed_vtracef(uint8_t dlevel, const char *grp, const char *fmt, va_list ap)
         mbed_trace_reset_tmp();
         goto end;
     }
-    if ((m_trace.trace_config & TRACE_MASK_LEVEL) &  dlevel) {
-        bool color = (m_trace.trace_config & TRACE_MODE_COLOR) != 0;
-        bool plain = (m_trace.trace_config & TRACE_MODE_PLAIN) != 0;
-        bool cr    = (m_trace.trace_config & TRACE_CARRIAGE_RETURN) != 0;
+    uint8_t config = mbed_trace_get_module_config(grp);
+    if ((config & TRACE_MASK_LEVEL) & dlevel) {
+        bool color = (config & TRACE_MODE_COLOR) != 0;
+        bool plain = (config & TRACE_MODE_PLAIN) != 0;
+        bool cr    = (config & TRACE_CARRIAGE_RETURN) != 0;
```

Comment:
Note that the `config` byte contains all of the above configurations. If this is intended to be "group specific config", that should be reflected in the setting function's name, or alternatively really only set the trace level with it.

Reply:
TBH I don't have strong opinion on that part. The trace level is the minimum that config should contain. For the other options what do you think would be the best ?

Reply:
I think it might be better to just restrict this to only the trace level. If you consider the other configuration bits they don't make much sense outside of global configuration. (Or at least I don't see much of one.)